src/main/java/org/jabref/JabRefExecutorService.java

```diff
@@ -147,6 +119,31 @@ public void shutdownEverything() {
         // kill the remote thread
         stopRemoteThread();
         // timer doesn't need to be canceled as it is run in daemon mode, which ensures that it is stopped if the application is shut down
+
+        Globals.taskExecutor.shutdown();
```

Comment:
With this statement you are creating a dependency from this relatively independent class to the gui package. I really do not like this. I think the shutting down of the gui-related `taskExecutor` should not happen in this class, especially since this one line is really the only position where it interacts with `JabRefExecutorService`. This is not cohesive.